packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -18,6 +18,11 @@ import { ContextManager, Context } from '@opentelemetry/api';
 import { EventEmitter } from 'events';
 
 type Func<T> = (...args: unknown[]) => T;
+type UnknownFunc = Func<unknown>;
+type UnknownFuncReturns<T extends (...args: unknown[]) => unknown> = (...args: unknown[]) => ReturnType<T>;
+
+const isUnknownFunc = (fn: unknown): fn is UnknownFunc => typeof fn === 'function';
+const isGenericFunc = <T>(fn: (...args: never[]) => T): fn is Func<T> => typeof fn === 'function';
```

Comment:
i don't see why we should differenciate between both, i think the second typeguard is fine to use everywhere no ?